NeuroID/NeuroIDClass/Extensions/NIDUser.swift

```diff
@@ -170,4 +206,26 @@ public extension NeuroID {
         }
         return true
     }
+    
+    internal static func scrubIdentifier(identifier: String) -> String {
+        do {
+            let emailRegex = try NSRegularExpression(pattern: "[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}")
+            let ssnRegex = try NSRegularExpression(pattern: "\\b\\d{3}-\\d{2}-\\d{4}\\b")
+            var result = emailRegex.matches(in: identifier, range: NSMakeRange(0, identifier.count))
+            if !result.isEmpty {
+                let atIndex = identifier.firstIndex(of: "@") ?? identifier.index(before: identifier.endIndex)
+                let idLength = identifier.distance(from: identifier.startIndex, to: atIndex)
+                let scrubbedEmailId = String(repeating: "*", count: idLength) + identifier[atIndex...]
+                return scrubbedEmailId
```

Comment:
I scrubbed all characters before `@` unlike android which leaves the first and last chars before `@`. I did this incase the email id is only 2 characters eg.  px@gmail.com would be **@gmail.com, but in android it would not be masked.

Reply:
we should do this on Android as well.

Reply:
clarified with Kevin and we will leave the first character to distinguish between multiple ids.